table/server/common/src/main/java/alluxio/table/common/CatalogPathUtils.java

```diff
@@ -85,8 +85,13 @@ public static AlluxioURI getFragmentsPath(String dbName, AlluxioURI fragmentUri)
    * @return escaped fragment URI
    */
   public static String escapeFragmentUfsPath(AlluxioURI fragmentUfsUri) {
-    // TODO(bowen): implement a real escaping algorithm
-    // e.g. path = path.replaceAll("/", "_")
-    return fragmentUfsUri.toString();
+    String scheme = fragmentUfsUri.getScheme();
+    scheme = scheme == null || scheme.isEmpty() ? "none" : scheme;
+    String authority = fragmentUfsUri.getAuthority().toString();
+    authority = authority == null || authority.isEmpty() ? "none" : authority;
+    String escapedFragmentPath = String.format("%s_%s", scheme, authority)
+        .replaceAll(":", "_")
+        .replaceAll("/", "_");
+    return escapedFragmentPath;
```

Comment:
This is to make sure we have some kind of identifier to distinguish these fragments. Since we mount the root of each UFS, the path leading to the tables is no longer relevent.